pkg/services/ngalert/accesscontrol.go

```diff
@@ -204,16 +204,37 @@ var (
 		},
 	}
 
+	routesReaderRole = accesscontrol.RoleRegistration{
+		Role: accesscontrol.RoleDTO{
+			Name:        accesscontrol.FixedRolePrefix + "alerting.routes:reader",
+			DisplayName: "Notification Policies Reader",
+			Description: "Read all notification policies in Grafana alerting",
+			Group:       AlertRolesGroup,
+			Permissions: accesscontrol.ConcatPermissions([]accesscontrol.Permission{
+				{Action: accesscontrol.ActionAlertingRoutesRead},
+			}),
+		},
+	}
+
+	routesWriterRole = accesscontrol.RoleRegistration{
+		Role: accesscontrol.RoleDTO{
+			Name:        accesscontrol.FixedRolePrefix + "alerting.routes:writer",
+			DisplayName: "Notification Policies Writer",
+			Description: "Update and reset notification policies in Grafana alerting",
+			Group:       AlertRolesGroup,
+			Permissions: accesscontrol.ConcatPermissions(routesReaderRole.Role.Permissions, []accesscontrol.Permission{
+				{Action: accesscontrol.ActionAlertingRoutesWrite},
+			}),
+		},
+	}
+
 	notificationsReaderRole = accesscontrol.RoleRegistration{
 		Role: accesscontrol.RoleDTO{
 			Name:        accesscontrol.FixedRolePrefix + "alerting.notifications:reader",
 			DisplayName: "Notifications Reader",
 			Description: "Read notification policies and contact points in Grafana and external providers",
 			Group:       AlertRolesGroup,
-			Permissions: accesscontrol.ConcatPermissions(receiversReaderRole.Role.Permissions, templatesReaderRole.Role.Permissions, timeIntervalsReaderRole.Role.Permissions, []accesscontrol.Permission{
-				{
-					Action: accesscontrol.ActionAlertingNotificationsRead,
-				},
```

Comment:
FYI removing this means that I can't call and get the alertmanager config anymore. Could/should we separate this change out from the PR? So we separately remove the access to the config endpoint unless you're an admin? (but FWIW, my admin user on local couldn't fetch the config endpoint still)

Reply:
Good point I added those actions back.